File: leetcode/0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        val_idx_dic = {}
        n = len(nums)
        for i in range(n):
            complement = target - nums[i]
            if complement in val_idx_dic:
                return [i, val_idx_dic[complement]]
            val_idx_dic[nums[i]] = i
                   
            
        

File: leetcode/0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> valIdxMap = new HashMap<>();
        for (int i =0; i< nums.length; i++) {
            int complement = target - nums[i];
            if (valIdxMap.containsKey(complement)) {
                return new int[]{i, valIdxMap.get(complement)};
            }
            valIdxMap.put(nums[i], i);
        }
        return new int[]{};
    }
}